### PR TITLE
feature: href for logo (with optional target="_blank")

### PR DESCRIPTION
**Pull request by breftejk**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by mcollina** [comment]
the tests are failing

File: types/index.d.ts [hunk 1]
```diff
@@ -78,6 +78,10 @@ declare namespace fastifySwaggerUi {
   type FastifySwaggerUILogo = {
     type: string;
     content: string | Buffer;
+    href?: {
+      url: string;
+      target_blank?: boolean;
+    }
```
Comment (Uzlopak) [inline types/index.d.ts:84]: I would actually say, that the logo gets a href and a target and not an object

```suggestion
    href?: string;
    target?: '_blank' | '_parent' | '_self' | '_top';
```

File: lib/swagger-initializer.js [hunk 2]
```diff
@@ -5,6 +5,7 @@ const serialize = require('./serialize')
 function swaggerInitializer (opts) {
   const logoBase64 = Buffer.from(opts.logo.content).toString('base64')
   const logoData = `data:${opts.logo.type};base64,${logoBase64}`
+  const logoHref = opts.logo.href
```
Comment (Uzlopak) [inline lib/swagger-initializer.js:8]: const logoHref = opts.logo.href
const logoTarget = opts.logo.target

File: lib/swagger-initializer.js [hunk 3]
```diff
@@ -62,6 +63,10 @@ function swaggerInitializer (opts) {
         const img = document.createElement('img')
         img.height = 40
         img.src = logoData
+        if(logoHref) {
+          img.href = logoHref.url
+          logoHref.target_blank && img.target = "_blank"
+        }
```
Comment (Uzlopak) [inline lib/swagger-initializer.js:69]: Would prefer this:

```suggestion
        img.href = logoHref || resolveUrl('/')
        logoTarget && (img.target = logoTarget)
```

**Review by Uzlopak — DISMISSED** [review]
I would make it actually simpler

File: README.md [hunk 4]
```diff
@@ -244,7 +244,11 @@ It's possible to override the logo displayed in the top bar by specifying:
 await fastify.register(require('@fastify/swagger-ui'), {
   logo: {
     type: 'image/png',
-    content: Buffer.from('iVBOR...', 'base64')
+    content: Buffer.from('iVBOR...', 'base64'),
+    href: {
+      url: '/documentation',
+      target_blank: true,
+    }
```
Comment (Uzlopak) [inline README.md:251]: can you fix this too, plz?

Comment (breftejk) [reply README.md:251]: I am literally just doing it, sec

**Comment by breftejk** [comment]
What is with this test error? I dont get it. Everything's fine on my test yet still this weird error on GitHub.

**Comment by climba03003** [comment]
Ignore the `pnpm` one, but `e2e` must be pass.

**Comment by Uzlopak** [comment]
@climba03003 PTAL

**Review by climba03003 — APPROVED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**